ui/src/contexts/Identity.tsx

```diff
@@ -0,0 +1,157 @@
+import { message } from "antd";
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+
+import { getIdentities } from "src/adapters/api/identities";
+import { useEnvContext } from "src/contexts/Env";
+import { AppError, Identity } from "src/domain";
+import { ROUTES } from "src/routes";
+import { AsyncTask, isAsyncTaskDataAvailable } from "src/utils/async";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+import { IDENTIFIER_SEARCH_PARAM, ROOT_PATH } from "src/utils/constants";
+import { buildAppError } from "src/utils/error";
+
+type IdentityState = {
+  fetchIdentities: (signal: AbortSignal) => void;
+  handleChange: (identifier: string) => void;
+  identifier: string;
+  identitiesList: AsyncTask<Identity[], AppError>;
+  identityDisplayName: string;
```

Comment:
Why do we need this property if it's part of the selected identity?

Reply:
we don't have selected identity, we have selected identifier instead

Reply:
Storing two properties that can potentially be out of sync it's not a good idea IMO. I see 2 options here:

  1. We store a full "selected identity" in the state, so anyone can use it directly.
  2. We store only the identifier and have a helper function that provided an identifier returns the selected identity.

I prefer 1.

Reply:
Agreed on a meet.